src/listeners/listener.discovery.ts

```diff
@@ -0,0 +1,28 @@
+import { NecordBaseDiscovery } from 'necord';
+import { NecordLavalinkEvents } from './listener.interface';
+
+export interface ListenerMeta {
+	type: 'once' | 'on';
+	event: NecordLavalinkEvents;
+}
+
+/**
+ * Represents a listener discovery.
+ */
+export class ListenerDiscovery extends NecordBaseDiscovery<ListenerMeta> {
+	public getType() {
+		return this.meta.type;
+	}
+
+	public getEvent() {
+		return this.meta.event;
+	}
+
+	public isListener(): this is ListenerDiscovery {
+		return true;
+	}
+
+	public override toJSON(): Record<string, any> {
+		return this.meta;
+	}
+}
```

Comment:
Btw, I think it will be not required to create discoverable classes. We can make logic more simpler via `Reflector.createDecorator()` like [here](https://github.com/necordjs/necord/blob/master/src/listeners/decorators/custom-listener.decorator.ts). Also explorer it via DiscoveryService, you can look to example:

Reply:
[Class-based example](https://github.com/necordjs/necord/blob/8a2df6b9b91957b9d873812f6cd542814d55552a/src/listeners/listeners.module.ts#L36)

![image](https://github.com/user-attachments/assets/df4c77d3-76f0-4e82-8d24-71767dae4cb1)

Reply:
But I prefer method-base decorator through all projects. You just copy that and remove part of filtering by `metadatakey: CustomListener.KEY`

Reply:
I can't see how to implement this